sigutils/types.h

```diff
@@ -103,8 +120,6 @@
 
 #define SU_C_ABS    SU_ADDSFX(cabs)
```

Comment:
All these macros (`SU_C_ABS`, `SU_C_EXP` and so on) will not work in C++ as they depend on functions like `cabs` or `cexp` which are not exposed, so they should be reimplemented too (either using the corresponding methods -`.abs()`- or functions -`std::exp(std::complex<float>)`-). You can find these methods/functions here in https://en.cppreference.com/w/cpp/numeric/complex

Reply:
Thanks. Here are my comments:

1. `abs`, `exp`, etc are not member functions, so we can't call the `.abs()` method.
We can either use `abs` or `std::abs` and it works fine in both cases. The same applies to `arg` and `exp`

2. `conj` works as is. Also `std::conj` works.

3. The only one that's tricky is the `SU_C_SGN` which throws a compile time error, as we're multiplying an integer and a complex and the `*` does not seem to be overloaded for that purpose.
So probably I should define it to something like
`#define SU_C_SGN(x)  SUCOMPLEX(SU_SGN(SU_C_REAL(x)), SU_SGN(SU_C_IMAG(x)))`
for C++

I will test and send the fix tonight.

Reply:
Since we don't know anything about the namespace this macros may be called from, I'd prefer to use `std::conj` here to prevent ambiguities due to name collisions. Apart from that, your proposal seems fine.